### ISSUE TITLE
Missing ICU data?

### ISSUE THREAD

**Issue by Laurian** [camertron/cldr-segmentation.js#1]
Hi,

I'm looking into having more or less the same results with https://www.npmjs.com/package/node-word-boundaries which is based on icu4c.

I was testing with Japanese: 

```
const word_boundaries = require('node-word-boundaries')
const cldrSegmentation = require('cldr-segmentation')

const text = '地図の裏には、あなたが毎日歩いているかもしれない場所についての歴史的な情報がありますが、気づくことはありません。ここでは、一番豊かな社会が、不幸なホースでスキルを共有する建物の外にあります。どうやら建物には幽霊がいるようだ。'

// node-word-boundaries
const boundaries = word_boundaries.find_word_boundaries(text)
const parts = word_boundaries.split(text)
console.log(parts) 

// cldr-segmentation
console.log(cldrSegmentation.wordSplit(text))
```

*node-word-boundaries* would give me 
```
[ '地図',
  'の',
  '裏',
  'に',
  'は',
  '、',
  'あなた',
  'が',
  '毎日',
  '歩',
  'い',
  'て',
  'いるか',
  'もし',
  'れ',
  'ない',
  '場所',
  'について',...
```
 while *cldr-segmentation* just:
```
[ '地',
  '図',
  'の',
  '裏',
  'に',
  'は',
  '、',
  'あ',…

  '不',
  '幸',
  'な',
  'ホース',
  'で',
  'スキル',
  'を',
  '共',…
```

Is there any ICU data that added somehow to your library would make things as full ICU?

**Comment by camertron** [comment]
Hey @laurian, unfortunately languages like Japanese, Chinese, Thai, etc are difficult to segment into words because they don't use spaces or other delimiting characters to indicate word boundaries. ICU'S implemention is derived from a bunch of dictionary data that I haven't looked into co-opting for this library. I have a feeling that, if included in cldr-segmentation, ICU dictionary data would be much too large to load in the browser. The implementation present in this library is based on regular expressions, which might not be compatible with the dictionary data from ICU. That said, I'd like to be able to support every language. If you've got some time, I'd welcome a pull request :)

**Comment by camertron** [comment]
This looks useful: http://userguide.icu-project.org/boundaryanalysis#TOC-Details-about-Dictionary-Based-Break-Iteration

**Comment by Laurian** [comment]
For Japanese I found https://github.com/leungwensen/tiny-segmenter to be very close to node-word-boundaries/icu4c results. I'll let you know what I'll find for Chinese and Thai.

**Comment by camertron** [comment]
Wow that's very cool!

**Comment by camertron** [comment]
Hey @Laurian, I just released v2.0 of this lib. While there's still no support for dictionary-based segmentation, it uses a new break engine based on the state machine in ICU4J/ICU4C. For that reason you may see more consistent results.
